GiniVision/Classes/Networking/MultipageDocumentsService.swift

```diff
@@ -29,41 +32,47 @@ final class MultipageDocumentsService: DocumentServiceProtocol {
             deleteCompositeDocument(withId: compositeDocument.documentId)
         }
         
+        analysisCancellationToken?.cancel()
+        analysisCancellationToken = nil
         compositeDocument = nil
     }
     
     func remove(document: GiniVisionDocument) {
         if let index = partialDocuments.index(forKey: document.id) {
             if let partialDocumentId = partialDocuments[document.id]?
+                .info
                 .documentId {
                 deletePartialDocument(withId: partialDocumentId)
             }
             partialDocuments.remove(at: index)
         }
     }
     
-    func update(parameters: [String: Any], for document: GiniVisionDocument) {
-        self.partialDocuments[document.id]?.updateAdditionalParameters(with: parameters)
+    func update(imageDocument: GiniImageDocument) {
+        partialDocuments[imageDocument.id]?.info.rotationDelta = Int32(imageDocument.rotationDelta)
     }
     
-    func orderDocuments(givenVisionDocumentIds ids: [String]) {
-        for index in 0..<ids.count {
-            let id = ids[index]
+    func sortDocuments(withSameOrderAs documents: [GiniVisionDocument]) {
+        for index in 0..<documents.count {
+            let id = documents[index].id
             partialDocuments[id]?.order = index
         }
+
     }
     
     func upload(document: GiniVisionDocument,
                 completion: UploadDocumentCompletion?) {
         let cancellationTokenSource = BFCancellationTokenSource()
         let token = cancellationTokenSource.token
-        self.partialDocuments[document.id] = PartialDocumentInfo()
+        self.partialDocuments[document.id] =
+            PartialDocumentInfo(info: (GINIPartialDocumentInfo(documentId: nil, rotationDelta: 0)),
```

Comment:
Ok, so there's `PartialDocumentInfo ` and `GINIPartialDocumentInfo `. That's confusing!

Reply:
Why can't you just use one of them?

Reply:
Because I don't want to let `GINIPartialDocumentInfo` object know about the order, since it is not part of it.